pcdsdevices/lens.py

```diff
@@ -520,14 +521,21 @@ class LensStack(LensStackBase):
 
     """
 
-    def __init__(self, *args, path, **kwargs):
+    def __init__(self, x_prefix, y_prefix, z_prefix, z_offset, z_dir, E,
```

Comment:
I realize that making these more explicit in many places allows for instrospectability, but I think we intentionally left this out in many cases to leverage UCpt to avoid writing all of this out. Do we need to always write out the full extended prefix set from now on in all classes to let the test suite do its thing?

Reply:
`UnrelatedComponent` is supported directly in the best-effort instantiation code. It _does_ assume you don't do something funny in `__init__` with the prefix (see the diff on how I changed AT2L0 - I was previously setting `calc_prefix` explicitly and not allowing the kwarg, which meant that `UCpt` wouldn't remove the argument from the kwargs dict).

See `pcdsdevices.lxe` or `pcdsdevices.jet` for passing `UCpt` examples.

The explicitness here is required for `FormattedComponent`, primarily. I personally think that requiring those in `__init__` kwarg lists explicitly is a reasonable thing.

Reply:
Ahh! Okay that makes a lot of sense. Seems :+1:

Reply:
That said, hopefully I didn't screw up any of the `__init__`s here inadvertently ;)